packages/react/select/src/select.tsx

```diff
@@ -1269,8 +1269,13 @@ const SelectItem = React.forwardRef<SelectItemElement, SelectItemProps>(
 
     const handleSelect = () => {
       if (!disabled) {
-        context.onValueChange(value);
-        context.onOpenChange(false);
+        const startTransition = React.startTransition
+          ? React.startTransition
+          : window.requestAnimationFrame;
+        startTransition(() => {
+          context.onValueChange(value);
+          context.onOpenChange(false);
+        });
```

Comment:
This seems wrong. The value change handler should have the highest priority (higher than render/paint) to quickly respond to the user interaction. Thus, you can't use `requestAnimationFrame` here.

The `startTransition` also feels wrong, as this schedules the value change to the next re-render.